Update the float on the Spinner to `none`

mapk opened the pull request:
## Description
Fixes #18199. It appears that changing the float on the spinner component to `none` instead of `right` seems to work fine. 

## How has this been tested?
I tested the visual change in a few places like: 

- Uploading an image to a block.
- Switching to the Media Library from an Image block.
- Publishing a post.

Each instance there seemed to work just fine visually. I only tested in Firefox 71.0.

## Screenshots 

**Image block**

![Screen Shot 2019-12-26 at 5 15 35 PM](https://user-images.githubusercontent.com/617986/71495790-8e43a700-2804-11ea-9820-0263da734afe.png)

**Storybook** 

![spinner 2019-12-26 17_19_47](https://user-images.githubusercontent.com/617986/71495799-9996d280-2804-11ea-83e7-51537461413a.gif)


## Types of changes
Non-breaking CSS changes.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->.

mapk committed:
Fixes #18199. Set the float to none in the Spinner's CSS.

Removed instances of float:none; in other CSS rules since the component is already set for this.

Remove the float completely.

chrisvanpatten commented:
I would love this to get merged (I’m sick of overriding this in my own code) but there are a few places where the spinner needs to be floated, such as when it’s shown above the compatibility mode metaboxes during saving, which would need to be updated. There are also a bunch of cases where the float is removed via CSS in specific instances, so that CSS would end up being redundant.

mapk commented:
@chrisvanpatten, Thanks for the feedback! I checked the meta boxes and that seems to work still.

![Screen Shot 2019-12-30 at 9 09 41 AM](https://user-images.githubusercontent.com/617986/71592436-12d44500-2ae5-11ea-87bf-3a5ea73310d2.png)

I've also updated a couple other files that were overriding the float and removed that style. This still should be tested by others before merging though.

mapk commented:
Also tested in the link interface and it appears correctly there too.

![Screen Shot 2020-01-03 at 5 11 53 PM](https://user-images.githubusercontent.com/617986/71757673-45b36b80-2e4c-11ea-8847-615753952942.png)

mapk commented on packages/components/src/spinner/style.scss:
```diff
@@ -4,7 +4,7 @@
 	height: 18px;
 	opacity: 0.7;
-	float: right;
+	float: none;
```
Should I keep this as `float:none;` or just remove it altogether? @jasmussen any thoughts?

jasmussen replied:
Interesting, if the float does not appear to regress anything, definitely remove it altogether. Floats are the worst, and if there's truly no downside to removing it, out it goes.

A float left or right literally takes an item out of the flow of content, giving it a zero height footprint (with no margins), that would be the only thing to look for, the non floated thing pushing things down.

mapk replied:
Thanks @jasmussen ✋ I've removed the float completely now.

mapk commented:
After removing the float completely, this can use a bit more testing. I've tested on the link creation popover, the uploading of images and it all looks good still.

jasmussen reviewed (approved):
This is working well for me, code looks good and a float really should fix more things than it breaks. 

I tested the most obvious spinner locations. From a brief code search it looks like the spinner is used in these places:

- Block directory search results
- Media upload progress
- URL input
- Block placeholder
- Embeds loading
- Galleries loading
- Images loading
- Latest posts loading
- Metaboxes saving
- Featured images doing something
- Post publish panel saving 

Those are the things to look for.

chrisvanpatten commented:
Thanks @mapk! Excited to put this to use :)
